// Image Service Protobuf Interface
syntax = "proto3";


message ProtoImg {
  bool color = 1;
  bytes data = 2;
  int32 width = 3;
  int32 height = 4;
}


message ProtoImgRotateRequest {
  enum Rotation {
    NONE = 0;
    NINETY_DEG = 1;
    ONE_EIGHTY_DEG = 2;
    TWO_SEVENTY_DEG = 3;
  }
  Rotation rotation = 1;
  ProtoImg image = 2;
}


service ProtoImgService {
  rpc rotate_img(ProtoImgRotateRequest) returns (ProtoImg);
  rpc moving_avg_filter(ProtoImg) returns (ProtoImg);
}
